### PR TITLE
[WIP] Remove LAssailant

### LINKED ISSUE #14592: LAssailant is garbage
It causes mobs to hard del because it keeps a ref to a mob forever unless the mob that was attacked is destroyed. That isn't conjecture, I checked.

We don't really use it, as far as I know, for anything but stat logs, and before that I don't know why it ever existed.
It isn't useful.
I'll probably remove it from the stats site at some point anyways.

We need to remove the LAssailant var. Why am I making this issue? Because I don't want to deal with it tonight and I don't want to forget how garbage it is. *And maybe someone else will do it for me*

### PR DESCRIPTION
**Pull request by gbasood**
Practically unused feature woo
Resolves #14592
I've got all of this out except the only place it was ever actually used, slime code.
And minimise casualties objective which was always bad anyway.
Someone's gonna have to figure out a new way for slime taming to work.
Didn't it used to work based on who was in sight when slimes ate things?

If travis ever sees this PR it should fail to compile because it's intentionally broken until I/someone else changes how slimes work.

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -742,6 +742,7 @@ var/list/potential_theft_objectives=list(
...
 	if(blocked)
 		return 0
-	if(owner.kills.len>5)
-		return 0
-	return 1
\ No newline at end of file
+	// haha this doesnt do anything anymore
+	// if(owner.kills.len>5)
+	// 	return 0
+	return 1
```
Comment (Kurfursten) [inline code/game/gamemodes/objective.dm:748]: 🤔

Comment (gbasood) [reply code/game/gamemodes/objective.dm:748]: Kills var used LAssailant

**Comment by MadmanMartian** [comment]
Why was this closed?

**Comment by Shadowmech88** [comment]
>I've got all of this out except the only place it was ever actually used, slime code.
>And minimise casualties objective which was always bad anyway.
>Someone's gonna have to figure out a new way for slime taming to work.

**Pull request closed without merging**